tests/unit/GraphAlgorithms.cpp

```diff
@@ -178,9 +224,285 @@ BOOST_AUTO_TEST_CASE(GetBackedgesTest) {
+
+  revng_check(Regions.size() == 2);
+  revng_check(Regions[0][0] == NLG.LoopLatch);
+  revng_check(Regions[0][1] == NLG.SecondLoopHeader);
+  revng_check(Regions[1][0] == NLG.SecondLoopLatch);
+  revng_check(Regions[1][1] == NLG.LoopHeader);
+  revng_check(Regions[1][2] == NLG.SecondLoopHeader);
+  revng_check(Regions[1][3] == NLG.LoopLatch);
+
+  // Compute the Reverse Post Order.
+  llvm::SmallVector<NodeType *> RPOT;
+  using RPOTraversal = llvm::ReversePostOrderTraversal<NodeType *>;
+  llvm::copy(RPOTraversal(NLG.Entry), std::back_inserter(RPOT));
+  revng_check(RPOT[0] == NLG.Entry);
+  revng_check(RPOT[1] == NLG.LoopHeader);
+  revng_check(RPOT[2] == NLG.SecondLoopHeader);
+  revng_check(RPOT[3] == NLG.LoopLatch);
+  revng_check(RPOT[4] == NLG.SecondLoopLatch);
+  revng_check(RPOT[5] == NLG.Exit);
+
+  // Collect the candidate entries for the identified inner region.
+  auto EntryCandidatesInner = getEntryCandidates<NodeType *>(Regions[0]);
+  auto EntryCandidatesInnerVector = EntryCandidatesInner.takeVector();
+  revng_check(EntryCandidatesInnerVector[0].first == NLG.SecondLoopHeader);
+  revng_check(EntryCandidatesInnerVector[0].second == 1);
+
+  EntryCandidatesInner = getEntryCandidates<NodeType *>(Regions[0]);
+
+  // Compute the distance of each node from the entry node.
+  auto ShortestPathFromEntry = computeDistanceFromEntry(NLG.Entry);
+
+  // Perform the election of the `Entry` node.
+  NodeType *EntryInner = electEntry<NodeType *>(EntryCandidatesInner,
+                                                ShortestPathFromEntry,
+                                                RPOT);
+
+  llvm::dbgs() << "Elected entry: " << EntryInner->getIndex() << "\n";
+  revng_check(EntryInner == NLG.SecondLoopHeader);
+
+  // Collect the candidate entries for the identified outer region.
+  auto EntryCandidatesOuter = getEntryCandidates<NodeType *>(Regions[1]);
+  auto EntryCandidatesOuterVector = EntryCandidatesOuter.takeVector();
+  revng_check(EntryCandidatesOuterVector[0].first == NLG.LoopHeader);
+  revng_check(EntryCandidatesOuterVector[0].second == 1);
+
+  EntryCandidatesOuter = getEntryCandidates<NodeType *>(Regions[1]);
+
+  // Perform the election of the `Entry` node.
+  NodeType *EntryOuter = electEntry<NodeType *>(EntryCandidatesOuter,
+                                                ShortestPathFromEntry,
+                                                RPOT);
+
+  llvm::dbgs() << "Elected entry: " << EntryOuter->getIndex() << "\n";
+  revng_check(EntryOuter == NLG.LoopHeader);
+}
+
+BOOST_AUTO_TEST_CASE(SimplifyInliningNestedRegionsTest) {
+  // Create the graph.
+  using NodeType = ForwardNode<MyForwardNode>;
+  auto INLG = createINLGGraph<NodeType>();
+  using EdgeDescriptor = revng::detail::EdgeDescriptor<NodeType *>;
+  using EdgeSet = SmallSetVector<EdgeDescriptor>;
+  using BlockSet = SmallSetVector<NodeType *>;
+  using BlockSetVect = llvm::SmallVector<BlockSet>;
+
+  // Compute the backedges set.
```

Comment:
No dots at the end of comments

Reply:
Regarding this point, since I saw that scattered in the codebase there are different behaviors about this. What should I do with explanation comments that have more than one sentence? like:
```
// Roses are red.
// Violets are blue.
// Fer are faster, Merc one-two
```
or
```
// Roses are red.
// Violets are blue.
// Fer is faster, Merc one-two.
```
?

Reply:
No dots at the end of comments for single line comments only. For the rest, the second option you gave it's OK.

Reply:
Right, in the end i went with no dots for _single sentence_ (not _single line_) comments, which seemed more coherent with the rest of the codebase (even if there are some mixed uses). Multi sentence comments have dots.